families/track_and_trade/sawtooth_track_and_trade/handler.py

```diff
@@ -0,0 +1,633 @@
+            TTPayload.CREATE_RECORD: _create_record,
+            TTPayload.FINALIZE_RECORD: _finalize_record,
+            TTPayload.CREATE_RECORD_TYPE: _create_record_type,
+            TTPayload.UPDATE_PROPERTIES: _update_properties,
+            TTPayload.CREATE_PROPOSAL: _create_proposal,
+            TTPayload.ANSWER_PROPOSAL: _answer_proposal,
+            TTPayload.REVOKE_REPORTER: _revoke_reporter,
+        }
+
+        try:
+            txn_handler = handlers[action]
+        except KeyError:
+            raise Exception('unhandled action')
+
+        txn_handler(payload, signer, timestamp, state)
+
+
+def _verify_timestamp(timestamp):
+    system_time = round(time.time())
+
+    if timestamp > system_time:
+        raise InvalidTransaction(
+            'Timestamp must be less than system time')
+
+
+def _unpack_transaction(transaction):
+    header = TransactionHeader()
+    header.ParseFromString(transaction.header)
+
+    signer = header.signer_pubkey
+
+    payload = TTPayload()
+    payload.ParseFromString(transaction.payload)
+
+    action = payload.action
+    timestamp = payload.timestamp
+
+    return signer, action, timestamp, payload
+
+
+def _get_container(state, address):
+    entry = state.get([address])
+
+    if not entry:
+        return None
+
+    data = entry[0].data
+
+    namespace = address[6:8]
+
+    containers = {
+        Addressing.AGENT: AgentContainer,
+        Addressing.PROPERTY: (PropertyContainer
+                              if address[-4:] == '0000'
+                              else PropertyPageContainer),
+        Addressing.PROPOSAL: ProposalContainer,
+        Addressing.RECORD: RecordContainer,
+        Addressing.RECORD_TYPE: RecordTypeContainer,
+    }
+
+    container = containers[namespace]()
+    container.ParseFromString(data)
+
+    return container
+
+
+def _set_container(state, address, container):
+    addresses = state.set([
+        StateEntry(
+            address=address,
+            data=container.SerializeToString(),
+        )
+    ])
+
+    if not addresses:
+        raise InternalError('State Error')
+
+
+def _verify_agent(state, signer):
+    address = Addressing.make_agent_address(signer)
+    container = _get_container(state, address)
+
+    if container is None or all([
+            agent.public_key != signer
+            for agent in container.entries]):
+        raise InvalidTransaction('Must be registered')
+
+
+def _create_agent(payload, signer, timestamp, state):
+    LOGGER.debug('_create_agent')
+    txn = payload.create_agent
+    name = txn.name
+
+    address = Addressing.make_agent_address(signer)
+    container = _get_container(state, address)
+
+    # Nothing is at the agent's address. Create a container and put
+    # the agent in it.
+    if container is None:
+        agent = Agent(
+            public_key=signer,
+            name=name,
+            timestamp=timestamp,
+        )
+
+        container = AgentContainer()
```

Comment:
Just create the Container like you have on this line, and then don't return.

The for loop will just skip since entries is empty, you'll log agent creation properly (which you aren't doing now), and then you only have to create your `Agent` and extend `entries` in one place.

So:
```
if container is None:
    container = AgentContainer()

for agent in container.entries:

. . .
```